### PR TITLE
Rename defaultNamespace to prefix and make it optional

### LINKED ISSUE #2335: Rename defaultNamespace => prefix
## Tell us about the problem you're trying to solve
* Step 1 of the name space project is to have a simple way to allow a user to avoid table names colliding in the destination.

## Describe the solution you’d like
* Rename `defaultNamespace` to `prefix`. It is just going to be a prefix for stream names. If it evolves in the future, so be it. For now it is a name space for stream names.
* This field should be _optional_.

### PR DESCRIPTION
**Pull request by ChristopheDuong**
## What
*Describe what the change is solving*
Closes #2335

## How
*Describe the solution*
Rename defaultNamespace to prefix

### CODE REVIEW

File: airbyte-api/src/main/openapi/config.yaml [hunk 1]
```diff
@@ -1620,15 +1620,15 @@ components:
...
         - syncCatalog
         - status
       properties:
         connectionId:
           $ref: "#/components/schemas/ConnectionId"
-        defaultNamespace:
+        prefix:
           type: string
-          description: Name of the connection that will determine where the source data will be written in the destination
+          description: Prefix used along namespace and table names that will determine where the source data will be written in the destination
```
Comment (cgardens) [inline airbyte-api/src/main/openapi/config.yaml:1631]:
```suggestion
          description: Prefix that will be prepended to the name of each stream when it is written to the destination.
```

File: airbyte-api/src/main/openapi/config.yaml [hunk 2]
```diff
@@ -1595,17 +1595,17 @@ components:
     ConnectionCreate:
       type: object
       required:
-        - defaultNamespace
+        - prefix
```
Comment (cgardens) [inline airbyte-api/src/main/openapi/config.yaml:1598]: shouldn't this line be deleted? putting it in the required array makes it required. we want it to be optional. same of all other structs in this file.
```suggestion
```

File: airbyte-api/src/main/openapi/config.yaml [hunk 3]
```diff
@@ -1620,15 +1620,15 @@ components:
       type: object
       required:
         - connectionId
-        - defaultNamespace
+        - prefix
```
Comment (cgardens) [inline airbyte-api/src/main/openapi/config.yaml:1623]:
```suggestion
```

File: airbyte-api/src/main/openapi/config.yaml [hunk 4]
```diff
@@ -1647,7 +1647,7 @@ components:
       required:
         - connectionId
         - name
-        - defaultNamespace
+        - prefix
```
Comment (cgardens) [inline airbyte-api/src/main/openapi/config.yaml:1650]:
```suggestion
```

File: airbyte-api/src/main/openapi/config.yaml [hunk 5]
```diff
@@ -1973,7 +1973,7 @@ components:
...
```
Comment (cgardens) [inline airbyte-api/src/main/openapi/config.yaml:1976]:
```suggestion
```

**Review by cgardens — APPROVED** [review]
The API interface configuration is still marking the prefix field as required. That's not what we want, right? After that is fixed, it looks good. It looks like you're changing stuff on the FE too. Probably means @Jamakase doesn't need to do anything (other than review this PR)

File: airbyte-workers/src/main/java/io/airbyte/workers/protocols/airbyte/NamespacingMapper.java [hunk 6]
```diff
@@ -39,45 +39,37 @@
...
+    stream.withName(transformStreamName(stream.getName(), prefix));
   }
 
-  private static String transformStreamName(final String streamName, final String defaultNamespace) {
-    // Use the default namespace as a prefix for the moment to alter the stream name in the destination
-    if (defaultNamespace != null && !defaultNamespace.isEmpty()) {
-      // TODO chris: use defaultNamespace as a namespace/schema/dataset in destination
-      return defaultNamespace + streamName;
+  private static String transformStreamName(final String streamName, final String prefix) {
+    if (prefix != null && !prefix.isEmpty()) {
```
Comment (michel-tricot) [inline airbyte-workers/src/main/java/io/airbyte/workers/protocols/airbyte/NamespacingMapper.java:70]: Strings.isEmpty or smthing like that does the same

**Review by michel-tricot — APPROVED** [review]
% charles comments

**Pull request merged**